### ISSUE TITLE
linter: incorrect `eslint-plugin-unicorn(no-useless-spread)` reported for `Uint8Array`

### ISSUE THREAD

**Issue by samualtnorman** [oxc-project/oxc#7936]
### What version of Oxlint are you using?

0.15.2

### What command did you run?

oxlint

### What does your `.oxlint.json` config file look like?

not using one

### What happened?

code like
```js
[ ...new Uint8Array([ 1, 2, 3 ]) ].map(byte => byte.toString())
```
reports `eslint-plugin-unicorn(no-useless-spread): Using a spread operator here creates a new array unnecessarily.` even though the spread is necessary
